app/src/navigation/screens/ClaimScreen.tsx

```diff
@@ -112,17 +112,19 @@ const ClaimScreen: React.FC = () => {
 
   const onSave = async () => {
     setLoading(true);
-    await addClaim(claim.type, formState, selectedFileIds);
     if (isEmailClaim) {
       if (!isEmail(formState.email as string)) {
+        setLoading(false);
         return Alert.alert(
           AlertTitle.Warning,
           "Please type a valid email claim value in the input field."
         );
       }
-      setLoading(false);
+    }
+    await addClaim(claim.type, formState, selectedFileIds);
```

Comment:
Could u not put this in the block above, considering they both check for `if (isEmailClaim)` ?

Reply:
Not at the moment with how the logic is working.